src/hotspot/cpu/s390/register_s390.hpp

```diff
@@ -57,235 +56,172 @@ typedef VMRegImpl* VMReg;
-// for use in debuggers and such.
-
-#ifndef DONT_USE_REGISTER_DEFINES
-#define noreg ((Register)(noreg_RegisterEnumValue))
-
-#define Z_R0  ((Register)(Z_R0_RegisterEnumValue))
-#define Z_R1  ((Register)(Z_R1_RegisterEnumValue))
-#define Z_R2  ((Register)(Z_R2_RegisterEnumValue))
-#define Z_R3  ((Register)(Z_R3_RegisterEnumValue))
-#define Z_R4  ((Register)(Z_R4_RegisterEnumValue))
-#define Z_R5  ((Register)(Z_R5_RegisterEnumValue))
-#define Z_R6  ((Register)(Z_R6_RegisterEnumValue))
-#define Z_R7  ((Register)(Z_R7_RegisterEnumValue))
-#define Z_R8  ((Register)(Z_R8_RegisterEnumValue))
-#define Z_R9  ((Register)(Z_R9_RegisterEnumValue))
-#define Z_R10 ((Register)(Z_R10_RegisterEnumValue))
-#define Z_R11 ((Register)(Z_R11_RegisterEnumValue))
-#define Z_R12 ((Register)(Z_R12_RegisterEnumValue))
-#define Z_R13 ((Register)(Z_R13_RegisterEnumValue))
-#define Z_R14 ((Register)(Z_R14_RegisterEnumValue))
-#define Z_R15 ((Register)(Z_R15_RegisterEnumValue))
-
-#define Z_CR ((ConditionRegister)(Z_CR_ConditionRegisterEnumValue))
-#endif // DONT_USE_REGISTER_DEFINES
-
+constexpr ConditionRegister Z_CR = as_ConditionRegister(0);
 
 //=========================
 //===  Float Registers  ===
 //=========================
 
-// Use FloatRegister as shortcut
-class FloatRegisterImpl;
-typedef FloatRegisterImpl* FloatRegister;
-
 // The implementation of float registers for the z/Architecture.
 
-inline FloatRegister as_FloatRegister(int encoding) {
-  return (FloatRegister)(long)encoding;
-}
-
-class FloatRegisterImpl: public AbstractRegisterImpl {
- public:
+class FloatRegister {
+  int _encoding;
+public:
   enum {
     number_of_registers     = 16,
     number_of_arg_registers = 4
   };
 
-  // construction
-  inline friend FloatRegister as_FloatRegister(int encoding);
+  constexpr FloatRegister(int encoding = -1) : _encoding(encoding) {}
+  bool operator==(const FloatRegister rhs) const { return _encoding == rhs._encoding; }
+  bool operator!=(const FloatRegister rhs) const { return _encoding != rhs._encoding; }
+  const FloatRegister* operator->() const { return this; }
 
-  inline VMReg as_VMReg();
+  // construction
+  inline constexpr friend FloatRegister as_FloatRegister(int encoding);
 
   // accessors
-  int encoding() const                                {
-     assert(is_valid(), "invalid register"); return value();
-  }
+  constexpr int encoding() const { assert(is_valid(), "invalid register"); return _encoding; }
+  inline VMReg as_VMReg() const;
+  FloatRegister successor() const { return FloatRegister(encoding() + 1); }
 
-  bool  is_valid() const          { return 0 <= value() && value() < number_of_registers; }
-  bool is_volatile() const        { return (0 <= (value()&0x7F) && (value()&0x7F) <= 7); }
-  bool is_nonvolatile() const     { return (8 <= (value()&0x7F) && (value()&0x7F) <= 15); }
+  // tester
+  constexpr bool  is_valid()      const { return 0 <= _encoding && _encoding < number_of_registers; }
```

Comment:
Indentation is different than for the other register types. I suggest to adapt it, here.

Reply:
Done; I did some other formatting changes, Please let me know if they looks good, I can happily revert them otherwise.